CI failure: Deploy — HOME environment variable incorrectly set

## CI Failure Report

**Workflow:** Deploy
**Branch:** main
**Commit:** ab0401dfc74e0b8f3b8d9e45133e0be33a948d3c
**Run:** https://github.com/Action-Llama/agents/actions/runs/23415131675

## Failure Summary

The Deploy workflow failed because the `al push` command could not find the environment configuration file. The HOME environment variable is being set to an empty string, causing the al CLI to look for the config in the wrong location.

## Failed Job(s)

### deploy

**Error output:**
```
=== Final pre-deploy checks ===
HOME: 
Current directory: /home/runner/work/agents/agents
=== Final credential check ===
No credentials found at /home/runner/work/agents/agents/.action-llama/credentials
=== Starting deploy ===
Configuration error: Environment "prod" not found at /home/runner/work/agents/agents/.action-llama/environments/prod.toml. Run 'al env init prod' to create it, or 'al env list' to see available environments.
##[error]Process completed with exit code 1.
```

## Root Cause Analysis

The issue is in the Deploy step of `.github/workflows/deploy.yml` at line 139:

```yaml
HOME: ${{ env.HOME }}
```

This line attempts to set the HOME environment variable using `${{ env.HOME }}`, but `env.HOME` doesn't exist in the GitHub Actions workflow context. This results in HOME being set to an empty string.

Earlier in the workflow, the environment file is created at `~/.action-llama/environments/prod.toml` (which resolves to `/home/runner/.action-llama/environments/prod.toml`). However, when HOME is empty, the `al` CLI looks for the environment file relative to the current working directory instead, searching at `/home/runner/work/agents/agents/.action-llama/environments/prod.toml` where it doesn't exist.

## Suggested Fix

Remove the problematic HOME environment variable setting from the Deploy step. The step should look like:

```diff
      - name: Deploy
        env:
          # Set environment variables as fallbacks for credentials
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          ANTHROPIC_API_KEY: ${{ secrets.ANTHROPIC_API_KEY }}
          # Ensure proper Git SSH setup
          GIT_SSH_COMMAND: "ssh -o StrictHostKeyChecking=no -i ~/.ssh/deploy_key"
-         # Explicitly set HOME to ensure al CLI looks in the right place
-         HOME: ${{ env.HOME }}
        run: |
```

Alternatively, if explicitly setting HOME is required, use:
```yaml
HOME: /home/runner
```

## Additional Context

This appears to be a regression introduced in PR #47 which was attempting to fix credential debugging. The PR added comprehensive debugging but inadvertently broke the deployment by incorrectly setting the HOME variable.

---
*This issue was automatically created by the gh-actions-responder agent.*

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -145,8 +145,6 @@ jobs:
           ANTHROPIC_API_KEY: ${{ secrets.ANTHROPIC_API_KEY }}
           # Ensure proper Git SSH setup
           GIT_SSH_COMMAND: "ssh -o StrictHostKeyChecking=no -i ~/.ssh/deploy_key"
-          # Explicitly set HOME to ensure al CLI looks in the right place
-          HOME: ${{ env.HOME }}
         run: |
           echo "=== Final pre-deploy checks ==="
           echo "HOME: $HOME"

